src/EditorFeatures/CSharpTest/QuickInfo/SyntacticQuickInfoSourceTests.cs

```diff
@@ -34,6 +34,87 @@ public void Brackets_1()
             ExpectedContent("if (true)\r\n{"));
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public void ScopeBrackets_0()
+        {
+            TestInMethod(@"
+            if (true)
+            {
+                {
+                }$$
+            }
+",
+            ExpectedContent("{"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public void ScopeBrackets_1()
+        {
+            TestInMethod(@"
+            if (true)
+            {
+                // some
+                // comment
+                {
+                }$$
+            }
+",
+            ExpectedContent(
+@"// some
+// comment
+{"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public void ScopeBrackets_2()
+        {
+            TestInMethod(@"
+            if (true)
+            {
+                /* comment */
+                {
+                }$$
+            }
+",
+            ExpectedContent(
+@"/* comment */
+{"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public void ScopeBrackets_3()
+        {
+            TestInMethod(@"
+            if (true)
+            {
+                {
+                    // some
+                    // comment
+                }$$
+            }
+",
+            ExpectedContent(
+@"{
+    // some
+    // comment"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.QuickInfo)]
+        public void ScopeBrackets_4()
+        {
+            TestInMethod(@"
+            if (true)
+            {
+                {
+                    /* comment */
+                }$$
```

Comment:
And finally, can you vary the tests by what the block is contained in? Reading the test makes me think this might only work for blocks inside blocks inside methodblocks, but it should also work for blocks inside methodblocks. So in some of the cases maybe remove the if (true) block.